docs/primitives/services-and-apis-ts.md

```diff
@@ -0,0 +1,206 @@
+* `api(async (): Promise<Response> => {});` &ndash; when you only return a response.
+* `api(async (params: Params): Promise<void> => {});` &ndash; when you only respond with success/fail.
+* `api(async (): Promise<void> => {});` &ndash; when you need neither request nor response data.
+
+The `api` function is a generic function. 
+
+You can also pass the type arguments for the request and response objects to the `api` function which looks like this: `api<Params, Response>(async (params) => {});`
+
+This approach is simple but very powerful. It lets Encore use [static analysis](/docs/introduction#meet-the-encore-application-model)
+to understand the request and response schemas of all your APIs, which enables Encore to automatically generate API documentation, type-safe API clients, and much more.
+
+### Access controls
+
+When you define an API, you have the option to supply an `APIOptions` object as the first argument to `api`. 
+In the options object you can set how the endpoint can be accessed:
+
+* `{ access: "public" }` &ndash; defines a public API that anybody on the internet can call (this is the default value if no access field is set).
+* `{ access: "private" }` &ndash; defines a private API that is never accessible to the outside world. It can only be called from other services in your app and via cron jobs.
+* `{ access: "auth" }` &ndash; defines a public API that anybody can call, but requires valid authentication.
+
+You can optionally send in auth data to `public` and `private` APIs, in which case the auth handler will be used. When used for `private` APIs, they are still not accessible from the outside world.
+
+[//]: # (TODO: Add link to auth guide when it's ready.)
+[//]: # (For more on defining APIs that require authentication, see the [authentication guide]&#40;/docs/develop/auth&#41;.)
+
+### REST APIs
+Encore has support for RESTful APIs and lets you easily define resource-oriented API URLs, parse parameters out of them, and more.
+
+To create a REST API, start by defining an endpoint and specify the `method` and `path` fields in the `APIOptions` object.
+
+[//]: # (TODO: Add link to when it's ready.)
+[//]: # (Learn more in the [API schemas guide](/docs/develop/api-schemas#path-parameters).)
+
+To specify a placeholder variable, use `:name` and add a function parameter with the same name to the function signature. Encore parses the incoming request URL and makes sure it matches the type of the parameter.
+
+For example, if you want to have a `getBlogPost` endpoint that takes a numeric id as a parameter:
+
+```typescript
+
+
+// getBlogPost retrieves a blog post by id.
+const getBlogPost = api(
+  { method: "GET", path: "/blog/:id" },
+  async ({ id }: { id: number }): Promise<BlogPost> => { 
+	  // Use id to query database...
+  },
+);
+```
+
+You can also combine path parameters with body payloads. For example, if you want to have an `updateBlogPost` endpoint:
```

Comment:
Is this possible at the moment? To have both a path param and a body payload?

Reply:
Yes